ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/netty/Callbacks.java

```diff
@@ -44,12 +45,20 @@
 
   public void invokeDisconnect(
       final PeerConnection connection,
-      final DisconnectReason reason,
+      final Optional<DisconnectReason> reason,
       final boolean initatedByPeer) {
     disconnectCallbacks.forEach(
         consumer -> consumer.onDisconnect(connection, reason, initatedByPeer));
   }
 
+  public void invokeDisconnect(
+      final PeerConnection connection,
```

Comment:
Any reason we need both optional and non-optional forms?

Reply:
Just for convenience - there are some cases where this api is called with a reason directly, and some cases where you're passing on the reason that's already wrapped in an `Optional`.